package.json

```diff
@@ -1,16 +1,50 @@
 {
   "private": true,
-  "workspaces": [
-    "dataregistry",
-    "dataregistry-extension"
-  ],
+  "workspaces": {
+    "packages": [
+      "packages/*"
+    ]
+  },
   "scripts": {
-    "build": "tsc --build dataregistry dataregistry-extension && jlpm workspace @jupyterlab/dataregistry-extension run generateSchema",
-    "build:dev": "jlpm install && jlpm clean && jlpm run build && jlpm run link",
-    "lint":  "jlpm workspace @jupyterlab/dataregistry-extension lint",
-    "clean": "jlpm workspace @jupyterlab/dataregistry run clean && jlpm workspace @jupyterlab/dataregistry-extension run clean",
-    "build:watch": "tsc --build --watch  dataregistry dataregistry-extension",
-    "link": "jupyter labextension link dataregistry --no-build && jupyter labextension link dataregistry-extension",
-    "unlink": "jupyter labextension unlink dataregistry --no-build && jupyter labextension unlink dataregistry-extension"
-  }
+    "all": "jlpm run clean && jlpm run build && jlpm run build:jupyter",
+    "all:watch": "jlpm run clean && jlpm run build && jlpm run build:jupyter:watch",
+    "build": "jlpm run build:dev",
+    "build:dev": "jlpm install && jlpm run clean:packages && lerna link && jlpm run build:packages && jlpm run link:packages && jupyter labextension list",
+    "build:jupyter": "jupyter lab build || (jlpm run build:jupyter:remediate && jupyter lab build)",
+    "build:jupyter:remediate": "curr_dir=$PWD && (cd $(jupyter lab path | head -n1 | sed s/Application\\ directory:\\ *//)/staging && echo $PWD && echo '\"@jupyterlab:registry\" \"http://localhost:4873\"' >> .yarnrc && cat .yarnrc && YARN_REGISTRY=http://localhost:4873 node ./yarn.js install --non-interactive && cd $curr_dir) || cd $curr_dir",
```

Comment:
Nothing will go wrong here! :)

Reply:
Nope. Nothing could go wrong. It would be better if we only performed the "remediation" step when we encounter the npm registry error, but that would require parsing the error log, which seems a bit heavy handed atm. Assuming we encounter a non-registry error which is reproducible, the second `jupyter lab build` should simply generate the same error. Or at least that is the hope...